proxy/src/main/java/com/velocitypowered/proxy/server/ServerMap.java

```diff
@@ -84,6 +87,7 @@ public RegisteredServer register(ServerInfo serverInfo) {
       throw new IllegalArgumentException(
           "Server with name " + serverInfo.getName() + " already registered");
     } else if (existing == null) {
+      if (server != null) server.getEventManager().fire(new ServerRegisteredEvent(serverInfo));
```

Comment:
Use the [fireAndForget](https://jd.papermc.io/velocity/3.0.0/com/velocitypowered/api/event/EventManager.html#fireAndForget(java.lang.Object)) method instead of [fire](https://jd.papermc.io/velocity/3.0.0/com/velocitypowered/api/event/EventManager.html#fire(E)) in case you are not going to use the resulting CompletableFuture

Reply:
Oh I didnt even notice that fire had a return type my bad